pages/static/survey.js

```diff
@@ -0,0 +1,255 @@
+};
+
+Fez.prototype.init = function () {
+  const answerExpiration = localStorage.getItem(`survey_${this.name}`);
+  const surveyTimestamp = parseInt(answerExpiration);
+
+  // if the answer expired, or there is no saved answer
+  // (and is therefore a NaN when parsed)...
+  if (Number.isNaN(surveyTimestamp) || Date.now() > new Date(surveyTimestamp)) {
+    localStorage.removeItem(`survey_${this.name}`);
+
+    this.build();
+  }
+};
+
+Fez.prototype._attachListeners = function () {
+  const inputs = [...$$('[type=radio]'), ...$$('[type=button]')];
+
+  inputs.forEach((e) => {
+    e.addEventListener('click', (e) => {
+      const el = e.target;
+      let card = el;
+
+      while (card && !card.classList.contains('active')) {
+        card = card.parentNode;
+      }
+
+      const index = card.parentNode.children.indexOf(card);
+      const question = this.questions[index];
+
+      if (question.type === 'likert' || question.type === 'rating') {
+        question.answer = el.nextSibling.textContent;
+      } else if (question.type === 'open') {
+        question.answer = card.querySelector('textarea').value;
+      }
+
+      this.nextSlide();
+    });
+  });
+
+  dismiss.addEventListener('click', this.submit);
+
+  setTimeout(this.nextSlide, 500);
+};
+
+Fez.prototype.nextSlide = async function () {
+  const cards = form.querySelectorAll('.slide');
+  const currentCard = form.querySelector('.active');
+  let currentIndex = cards.indexOf(currentCard);
+  const nextIndex = (currentIndex += 1);
+
+  if (nextIndex < cards.length) {
+    if (currentCard) {
+      currentCard.classList.remove('active');
+    }
+
+    cards[nextIndex].classList.add('active');
+
+    if (nextIndex === cards.length - 1) {
+      dismiss.classList.add('hide');
+      this.submit();
+    }
+  }
+
+  const {height} = await cards[nextIndex].getBoundingClientRectAsync();
+
+  form.style.height = `${height}px`;
+};
+
+Fez.prototype.submit = async function (dismissed) {
+  $('#fez').style.height = 0;
+
+  try {
+    fetch(`${self.location.origin}/fez-survey-response`, {
+      method: 'POST',
+      cache: 'no-cache',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        'survey': this.name,
+        'questions': this.questions,
+        'url': this.url,
+        'shownAt': `${this.shownAt}`,
+        dismissed,
+      }),
+    });
+  } catch (e) {
+    // we don't actually care if there are errors at this point
+    // they will show up in the server logs, and we don't want
+    // to show any error to the end user.
+  }
+
+  // store the fact we replied to the survey for 4 weeks
+  localStorage.setItem(
+    `survey_${this.name}`,
+    Date.now(this.shownAt) + 1000 * 60 * 60 * 24 * 28
+  );
+
+  this.remove(dismissed);
+};
+
+Fez.prototype.remove = function (dismissed) {
+  form.style.height = 0;
+  setTimeout(
+    () => {
+      form.parentNode.innerHTML = '';
+    },
+    dismissed ? 0 : 2500
```

Comment:
2500?

Reply:
If it was dismissed, close it right away, if it is not (and was therefore submitted), wait 2500 milliseconds before hiding, in order to show the "thank you!"

Reply:
"Extract to a well-named variable" 😁